lib/adapters/call-hierarchy-adapter.ts

```diff
@@ -0,0 +1,145 @@
+import type * as atomIde from "atom-ide-base"
+import Convert from "../convert"
+import * as Utils from "../utils"
+import { SymbolTag } from "../languageclient"
+import type { LanguageClientConnection, ServerCapabilities, CallHierarchyItem } from "../languageclient"
+import type { CancellationTokenSource } from "vscode-jsonrpc"
+import type { Point, TextEditor } from "atom"
+
+import OutlineViewAdapter from "./outline-view-adapter"
+
+/** Public: Adapts the documentSymbolProvider of the language server to the Outline View supplied by Atom IDE UI. */
+export default class CallHierarchyAdapter {
+  private _cancellationTokens: WeakMap<LanguageClientConnection, CancellationTokenSource> = new WeakMap()
+
+  /**
+   * Public: Determine whether this adapter can be used to adapt a language server based on the serverCapabilities
+   * matrix containing a callHierarchyProvider.
+   *
+   * @param serverCapabilities The {ServerCapabilities} of the language server to consider.
+   * @returns A {Boolean} indicating adapter can adapt the server based on the given serverCapabilities.
+   */
+  public static canAdapt(serverCapabilities: ServerCapabilities): boolean {
```

Comment:
Please make this an exported function instead of a class method

Reply:
I think I fixed it with https://github.com/atom-community/atom-languageclient/pull/172/commits/48b62a5b5280fc5c7d623d5a0c5638ca76c5df72, but the test now fails on windows...😓
I don't know why.

Reply:
I will soon take look at it.